querier.hh

```diff
@@ -180,6 +200,24 @@ public:
         }
         return full_position_view(dk->key(), _compaction_state->current_position());
     }
+
+private:
+    sstring current_range_to_string() const {
+        sstring keystr = "<no key>";
+        std::optional<partition_key> key(std::nullopt);
+        if (const dht::partition_range *prange = _range.get()) {
+            if (const auto &start = prange->start()) {
+                key = start->value().key();
+            } else if (const auto &end = prange->end()) {
+                key = end->value().key();
+            }
+        }
+        if (key) {
+            keystr = format("{}", (*key).with_schema(*_schema.get()));
+            return utils::utf8::validate((const uint8_t *) keystr.data(), keystr.size()) ? keystr : "<non-utf8-key>";
+        }
+        return keystr;
```

Comment:
There is going to be two cases here w.r.t. to the form of `_range`:
* Point queries (reading a single partition): `_range` a singular range: `_range->singular() == true` and `_range->start()` is engaged and has a key. In this case you should simply return `format("{}", _range->start()->value().with_schema(schema));` which will return the nicely formatted partition key the query is reading.
* Partition range scan: `_range` is a token-interval, with no keys in either range. In this case this method should return the string "partition-range-scan" or similar. There is no point trying to print the actual `vnode` interval, it is of no use to the user.